File: src/wp/Fonts.cs
using System;
using System.Runtime.Serialization;

namespace WPCordovaClassLib.Cordova.Commands
{
    /// <summary>
    /// Base Fonts object
    /// </summary>
    public class Fonts : BaseCommand
    {
        /// <summary>
        /// Returns a string array containing the (an-us) names of the fonts installed on the system.
        /// </summary>
        /// <param name="options"></param>               
        public void getFontList(string options)
        {
            try
            {
                string[] fontFamilies = Microsoft.Graphics.Text.CanvasTextFormat.GetSystemFontFamilies();
                PluginResult result = new PluginResult(PluginResult.Status.OK, this.WrapIntoJSON(fontFamilies));
                this.DispatchCommandResult(result);
            }
            catch (Exception)
            {
                this.DispatchCommandResult(new PluginResult(PluginResult.Status.ERROR, new GlobalizationError()));
            }
        }
    }

    /// <summary>
    /// Wraps data into JSON format
    /// Method taken from Apache / cordova-plugin-globalization
    /// </summary>
    /// <param name="data">data</param>
    /// <returns>data formatted as JSON object</returns>
    static string WrapIntoJSON<T>(T data, string keyName = "value", string stringifiedData = null)
    {
        string param = "{0}";
        stringifiedData = stringifiedData ?? data.ToString();

        if (data.GetType() == typeof(string))
        {
            param = "\"" + param + "\"";
        }

        if (data.GetType() == typeof(bool))
        {
            stringifiedData = stringifiedData.ToLower();
        }

        if (data.GetType() == typeof(string[]))
        {
            stringifiedData = JsonHelper.Serialize(data);
        }

        var formattedData = string.Format("\"" + keyName + "\":" + param, stringifiedData);
        formattedData = "{" + formattedData + "}";

        return formattedData;
    }    
}

File: www/fonts.js
var exec = require('cordova/exec');

var Fonts = {

/**
* Returns a list of fonts installed on the device.
*
* @param {function} successCB
* @param {function} errorCB
*
* @return Object.value {Array{String}}: the list of fonts installed on this device.
* Example
*    Fonts.getFontList(function(fontList) {console.log(fontList);},
*                      function(error) {console.log(error);});
*/
    getFontList: function (successCB, errorCB) {
        exec(successCB, errorCB, "Fonts", "getFontList", []);
    },
/**
* Returns the string name of the default font on the device.
*
* @param {function} successCB
* @param {function} errorCB
*
* @return Object.value {String}: the default font name
*
* Example
*    Fonts.getDefaultFont(function(fontName) {console.log(fontName);},
*                      function(error) {console.log(error);});
*/
    getDefaultFont: function (successCB, errorCB) {
        exec(successCB, errorCB, "Fonts", "getDefaultFont", []);
    }

};

module.exports = Fonts;